### ISSUE TITLE
Could not find a JavaScript runtime

### ISSUE THREAD

**Issue by mcnesium** [middleman/middleman#2325]
On Manjaro Linux I tried to install and run middleman. Got the following errors when running `init`:

* cannot load such file -- coffee_script
* Could not find a JavaScript runtime. 

What am I missing?

<details>
    <summary>Detailed command line log</summary>

    19:30:54 mcn@x270 ~ gem install middleman
    Successfully installed middleman-4.3.6
    1 gem installed
    19:31:02 mcn@x270 ~ middleman init middleman_test
    Traceback (most recent call last):
        16: from /home/mcn/.local/share/gem/bin/middleman:23:in `<main>'
        15: from /home/mcn/.local/share/gem/bin/middleman:23:in `load'
        14: from /home/mcn/.local/share/gem/gems/middleman-cli-4.3.6/bin/middleman:49:in `<top (required)>'
        13: from /home/mcn/.local/share/gem/gems/middleman-cli-4.3.6/bin/middleman:49:in `new'
        12: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/application.rb:263:in `initialize'
        11: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/application.rb:263:in `new'
        10: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/extension_manager.rb:12:in `initialize'
        9: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/extensions.rb:125:in `load_settings'
        8: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/extensions.rb:125:in `each'
        7: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/extensions.rb:127:in `block in load_settings'
        6: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/extensions.rb:91:in `load'
        5: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/core_extensions/rendering.rb:11:in `block in <top (required)>'
        4: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
        3: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
        2: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/renderers/coffee_script.rb:2:in `<top (required)>'
        1: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require'
    /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:92:in `require': cannot load such file -- coffee_script (LoadError)
        23: from /home/mcn/.local/share/gem/bin/middleman:23:in `<main>'
        22: from /home/mcn/.local/share/gem/bin/middleman:23:in `load'
        21: from /home/mcn/.local/share/gem/gems/middleman-cli-4.3.6/bin/middleman:49:in `<top (required)>'
        20: from /home/mcn/.local/share/gem/gems/middleman-cli-4.3.6/bin/middleman:49:in `new'
        19: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/application.rb:263:in `initialize'
        18: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/application.rb:263:in `new'
        17: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/extension_manager.rb:12:in `initialize'
        16: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/extensions.rb:125:in `load_settings'
        15: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/extensions.rb:125:in `each'
        14: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/extensions.rb:127:in `block in load_settings'
        13: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/extensions.rb:91:in `load'
        12: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/core_extensions/rendering.rb:11:in `block in <top (required)>'
        11: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
        10: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
        9: from /home/mcn/.local/share/gem/gems/middleman-core-4.3.6/lib/middleman-core/renderers/coffee_script.rb:2:in `<top (required)>'
        8: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:156:in `require'
        7: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:168:in `rescue in require'
        6: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:168:in `require'
        5: from /home/mcn/.local/share/gem/gems/coffee-script-2.4.1/lib/coffee_script.rb:1:in `<top (required)>'
        4: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
        3: from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:72:in `require'
        2: from /home/mcn/.local/share/gem/gems/execjs-2.7.0/lib/execjs.rb:4:in `<top (required)>'
        1: from /home/mcn/.local/share/gem/gems/execjs-2.7.0/lib/execjs.rb:5:in `<module:ExecJS>'
    /home/mcn/.local/share/gem/gems/execjs-2.7.0/lib/execjs/runtimes.rb:58:in `autodetect': Could not find a JavaScript runtime. See https://github.com/rails/execjs for a list of available runtimes. (ExecJS::RuntimeUnavailable)
</details>

<details>
    <summary>List of installed gems</summary>

    19:31:13 mcn@x270 ~ gem list

    *** LOCAL GEMS ***

    activesupport (5.0.7.2)
    addressable (2.7.0)
    backports (3.17.2)
    benchmark (default: 0.1.0)
    bigdecimal (default: 2.0.0)
    bundler (2.1.4)
    cgi (default: 0.1.0)
    coffee-script (2.4.1)
    coffee-script-source (1.12.2)
    colorator (1.1.0)
    concurrent-ruby (1.1.6)
    contracts (0.13.0)
    csv (default: 3.1.2)
    date (default: 3.0.0)
    dbm (default: 1.1.0)
    delegate (default: 0.1.0)
    did_you_mean (default: 1.4.0)
    dotenv (2.7.5)
    em-websocket (0.5.1)
    erubis (2.7.0)
    etc (default: 1.1.0)
    eventmachine (1.2.7)
    execjs (2.7.0)
    fast_blank (1.0.0)
    fastimage (2.1.7)
    fcntl (default: 1.0.0)
    ffi (1.12.2)
    fiddle (default: 1.0.0)
    fileutils (default: 1.4.1)
    forwardable (default: 1.3.1)
    forwardable-extended (2.6.0)
    gdbm (default: 2.1.0)
    getoptlong (default: 0.1.0)
    haml (5.1.2)
    hamster (3.0.0)
    hashie (3.6.0)
    http_parser.rb (0.6.0)
    i18n (0.9.5)
    io-console (default: 0.5.6)
    ipaddr (default: 1.2.2)
    irb (1.2.1)
    jekyll (4.0.1)
    jekyll-feed (0.13.0)
    jekyll-sass-converter (2.1.0)
    jekyll-seo-tag (2.6.1)
    jekyll-watch (2.2.1)
    json (default: 2.3.0)
    kramdown (2.2.1, 1.17.0)
    kramdown-parser-gfm (1.1.0)
    liquid (4.0.3)
    listen (3.0.8)
    logger (default: 1.4.2)
    matrix (default: 0.2.0)
    memoist (0.16.2)
    mercenary (0.3.6)
    middleman (4.3.6)
    middleman-cli (4.3.6)
    middleman-core (4.3.6)
    minima (2.5.1)
    minitest (5.14.1)
    mutex_m (default: 0.1.0)
    net-pop (default: 0.1.0)
    net-smtp (default: 0.1.0)
    observer (default: 0.1.0)
    open3 (default: 0.1.0)
    openssl (default: 2.1.2)
    ostruct (default: 0.2.0)
    padrino-helpers (0.13.3.4)
    padrino-support (0.13.3.4)
    parallel (1.19.1)
    pathutil (0.16.2)
    prime (default: 0.1.1)
    pstore (default: 0.1.0)
    psych (default: 3.1.0)
    public_suffix (4.0.5)
    racc (default: 1.4.16)
    rack (2.2.2)
    rb-fsevent (0.10.4)
    rb-inotify (0.10.1)
    readline (default: 0.0.2)
    readline-ext (default: 0.1.0)
    reline (default: 0.1.3, 0.0.7)
    rexml (3.2.4, default: 3.2.3)
    rouge (3.19.0)
    rss (default: 0.2.8)
    safe_yaml (1.0.5)
    sassc (2.3.0)
    sdbm (default: 1.0.0)
    servolux (0.13.0)
    singleton (default: 0.1.0)
    stringio (default: 0.1.0)
    strscan (default: 1.0.3)
    temple (0.8.2)
    terminal-table (1.8.0)
    thor (1.0.1)
    thread_safe (0.3.6)
    tilt (2.0.10)
    timeout (default: 0.1.0)
    tracer (default: 0.1.0)
    tzinfo (1.2.7)
    uglifier (3.2.0)
    unicode-display_width (1.7.0)
    uri (default: 0.10.0)
    webrick (default: 1.6.0)
    yaml (default: 0.1.0)
    zlib (default: 1.1.0)
</details>

<details>
    <summary>Ruby and Gem Versions</summary>

    19:31:27 mcn@x270 ~ gem --version
    3.1.2
    19:31:35 mcn@x270 ~ ruby --version
    ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]

</details>

**Comment by tdreyno** [comment]
You'll need to select a JS runtime. Most OSes ship with one, but I guess manjaro doesn't 

You'll want to add `therubyracer` to your Gemfile, which is a copy of the Google V8 engine that we can use to interact with javascript. Its a beefy library and may require you to add additional packages to your own to compile and link it.

**Comment by tdreyno** [comment]
More info: https://github.com/rubyjs/therubyracer

**Comment by qichunren** [comment]
middleman 4.3.8, I got same errors when middleman init

**Comment by tdreyno** [comment]
No, this is good. This actually looks like a bug. It seems the CLI is trying to see if we're in a existing project, which because we're calling 'init', we definitely aren't. Let me take a look.

**Comment by phrequencyviii** [comment]
I'm getting the same thing on macOS Catalina. Tried to install therubyracer as well but it either failed or didn't fix the issue.

`/usr/local/lib/ruby/gems/2.7.0/gems/execjs-2.7.0/lib/execjs/runtimes.rb:58:in `autodetect': Could not find a JavaScript runtime. See https://github.com/rails/execjs for a list of available runtimes. (ExecJS::RuntimeUnavailable)`

**Comment by kcomandich** [comment]
I got this error on a new install of Big Sur. It cleared up for me when I installed node with homebrew.

**Comment by ciscodesign** [comment]
Here another +1. OS X, Big Sur, same problems

**Comment by tdreyno** [comment]
Should be the same solution. Install node via homebrew or nvm.

**Comment by ciscodesign** [comment]
Nope, doesn't solve :(

**Comment by jltml** [comment]
@ciscodesign not sure if you've already solved this, but just in case — I ran into an issue with JS runtimes when I upgraded to Big Sur and [here's how I fixed it](https://github.com/middleman/middleman/issues/2196#issuecomment-730626318), if that's somehow helpful? Then again, I just now removed `sq_mini_racer` (which I'd needed to get Middleman working again) after having installed Node with [`n`](https://github.com/tj/n) a couple weeks ago and everything still seems to be working fine, so I'm not sure if this would help. Either way... good luck!
